clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -2267,6 +2301,7 @@ protected DelegateConsumerRecordResult delegateConsumerRecord(
             } else {
               if (controlMessageType == START_OF_SEGMENT
                   && Arrays.equals(consumerRecord.getKey().getKey(), KafkaKey.HEART_BEAT.getKey())) {
+                recordHeartbeatReceived(partitionConsumptionState, consumerRecord, kafkaUrl);
```

Comment:
If we plan to use the monitored lag for data freshness then we should report this in the drainer (when data is getting persisted and PCS offset is getting updated) instead of when we receive it from the consumer right?

Reply:
I suppose we could, are heartbeats queued to drainer today?

I'm not sure if that would make too much difference.  If drainer is blocked, then consumption will be blocked, so we'll still get the lag/alert.

Reply:
Yeah, it depends if you want to use this lag metric as a lower bound or not.

Reply:
Put into the drainer